### ISSUE TITLE
Missing documentation on testing json body posts in integration tests

### ISSUE THREAD

**Issue by adamthedeveloper** [rails/rails#38913]
### Steps to reproduce
Search through the testing portion of the guide [https://edgeguides.rubyonrails.org/testing.html](https://edgeguides.rubyonrails.org/testing.html) for an example on how to test a post with a json body.

The result will be that you won't find an example that makes it clear on how to do this. When initializing a rails app, you have an option to build it as an api. So, I'd assume that I'd also learn how to test json body posts.

### Expected behavior
Have realistic documentation on posting JSON data to a Rails 6 API. I can't find anything at all on how to do this in the guide: [https://edgeguides.rubyonrails.org/testing.html](https://edgeguides.rubyonrails.org/testing.html)

### Actual behavior
Not able to write a post test that uses a json post body because I simply cannot see how to do it.

### System configuration
**Rails version**:
latest
**Ruby version**:
any ruby version

**Comment by adamthedeveloper** [comment]
Here is where the documentation gets very misleading:

```
The get method kicks off the web request and populates the results into the @response. It can accept up to 6 arguments:

The URI of the controller action you are requesting. This can be in the form of a string or a route helper (e.g. articles_url).
params: option with a hash of request parameters to pass into the action (e.g. query string parameters or article variables).
headers: for setting the headers that will be passed with the request.
env: for customizing the request environment as needed.
xhr: whether the request is Ajax request or not. Can be set to true for marking the request as Ajax.
as: for encoding the request with different content type. Supports :json by default.
```

Notice, you can add the as: key to the get method according to the documentation. Turns out, it also works on the post call as well. Can someone change the documentation so that the as: doesn't seem to apply only to the get method?

**Comment by adamthedeveloper** [comment]
Also, the comment "Supports :json by default" makes no sense. Especially if you have to specify it in the post options.

**Comment by chaadow** [comment]
You don't have to specify `as: :json` that's why it says `supports :json by default`, also if you keep reading the same section, you have examples for patch and post.

Regarding the `get` paragraph, it's just an explanation for the code snippet above it, but you have more examples demonstrating the options described if you keep reading:

```ruby
patch article_url(Article.last), params: { article: { title: "updated" } }, xhr: true
```
Or this:
```ruby
post articles_url, params: { article: { body: 'Rails is awesome!', title: 'Hello Rails' } }, headers: { Authorization: ActionController::HttpAuthentication::Basic.encode_credentials('dhh', 'secret') }
```

However what's missing ( perhaps I haven't carefully perused the section) is an example with the `env` option
One example I've personally used in my apps is to override the `QUERY_STRING` for a SaaS app. like so : 
```ruby
post articles_url, env: { 'QUERY_STRING' => "_tenant=test-tenant" }
```

**Comment by eugeneius** [comment]
> Can someone change the documentation

Yes: you can! 🙂 If you think you can make it clearer, please do send a pull request.

I agree that "Supports `:json` by default" is potentially confusing. The API documentation covers how the `as:` option works in much more detail:

https://api.rubyonrails.org/classes/ActionDispatch/IntegrationTest.html#class-ActionDispatch::IntegrationTest-label-Changing+the+request+encoding

That level of detail is inappropriate for the guides though; maybe we should remove the explicit mention of JSON support in favour of an example that makes a POST request with `as: :json`.

**Comment by danieldimitrov** [comment]
Since I didn't see any activity here, I gave it a shot. I hope it is OK.

> That level of detail is inappropriate for the guides though; maybe we should remove the explicit mention of JSON support in favour of an example that makes a POST request with as: :json.

I've updated guides based on this. Hopefully, it is more clear now.

**Comment by eugeneius** [comment]
Closed by https://github.com/rails/rails/pull/39075. Thanks @danieldimitrov!
